### DISCUSSION TITLE
celery worker don't support read dags from db?

### DISCUSSION THREAD

**Discussion by luoyuliuyin** [apache/airflow#15560] [Q&A]
I want to create a dag, the following are my steps:

1. I create a dag name `a_weixiang_test ` in the `dags_folder` directory (the dag file only create in scheduler, not sync to web and worker).
![image](https://user-images.githubusercontent.com/28948186/116346707-58aa3600-a81d-11eb-8e4b-06a4c67fe7dc.png)

2. Then access the dag through ui, i can visit the dag. This means that the webServer can get dag through db.
![image](https://user-images.githubusercontent.com/28948186/116346414-cd30a500-a81c-11eb-9b14-4ec6c78d511e.png)

3. Trigger a new dag run via ui, i get the following error message:
`Failed to execute task dag_id could not be found: a_weixiang_test. Either the dag did not exist or it failed to parse..`
![image](https://user-images.githubusercontent.com/28948186/116346239-6f9c5880-a81c-11eb-9c93-8a1290bde2d8.png)

I know that as long as I sync the dag file to the `dags_folder` directory of the worker, all process can work well，
but i don't want to do this, i hope the worker can get dag from db, can this be done?

I noticed that the data in the database does not meet expectations, all pickle-related fields of `dag` table are empty, these fields are: `last_pickled`, `last_expired`, `pickle_id`,but the `dag_pickle` have records, does this matter?
![image](https://user-images.githubusercontent.com/28948186/116346551-17b22180-a81d-11eb-8fd4-3d2c65a21f84.png)
![image](https://user-images.githubusercontent.com/28948186/116349032-b6408180-a821-11eb-9863-d1d5f00d3062.png)

My os version: `centos 7.2`
My airflow version: `2.0.1`

My start command：
`airflow webserver`
`airflow scheduler --do-pickle`
`airflow celery worker`

My environment configuration:
`executor=CeleryExecutor`
`broker_url=amqp://user:password@xxx:5672`
`result_backend=db+mysql://user:password@xxx:3306/airflow`
`sql_alchemy_conn=mysql://user:password@xxx:3306/airflow`
`donot_pickle=False`
`store_dag_code = True`

**Comment by PxTruy** [comment]
@luoyuliuyin Have you found a solution to this problem ？

**Reply by luoyuliuyin** [reply]
No, I think this is caused by Airflow's plan to abandon Pickler. Now airflow uses json to achieve serialization, and WEBSERVER implements reading data from db and deserializing it, the WORKER has not yet completed this part of the work.

**Reply by PxTruy** [reply]
I see.

**Reply by handol-park** [reply]
Does worker still not support this? @luoyuliuyin

**Reply by luoyuliuyin** [reply]
yes, now workers only support local dag, pickler mode is about to be abandoned for security reasons

**Reply by potiuk** [reply]
Yep. Picke does not work. Forget about it.

**Reply by dylan-turnbull** [reply]
It seems the documentation has also been abandoned. Forget about it.
